go-controller/pkg/util/ndp/ra.go

```diff
@@ -0,0 +1,101 @@
+package ndp
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/mdlayher/socket"
+	"golang.org/x/sys/unix"
+)
+
+// RouterAdvertisement with mac, ips and lifetime field to send
+type RouterAdvertisement struct {
+	SourceMAC, DestinationMAC net.HardwareAddr
+	SourceIP, DestinationIP   net.IP
+	Lifetime                  uint16
+}
+
+// SendRouterAdvertisements will send variadic RAs.
+// Requires raw socket capabilities, since the RAs src MAC
+// and IP are not the ones on the LRP.
+func SendRouterAdvertisements(interfaceName string, ras ...RouterAdvertisement) error {
+	iface, err := net.InterfaceByName(interfaceName)
+	if err != nil {
+		return err
+	}
+	c, err := socket.Socket(syscall.AF_PACKET, syscall.SOCK_RAW, syscall.ETH_P_ALL, "ra", nil)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+	serializedRAs := [][]byte{}
```

Comment:
this logic and below - was it inspired from the ndp library? I dunno enough to  check its correctness

Reply:
So I came up with thise using the ipv6toolkit to generate the unsolicited RAs that make the default gw multipath be reconciled, then used tcpdump to inspect them, then replicate whatever is needed with this library comparing tcpdumps.

The only validation we have that is ok is the e2e, ipv6 default gw is reconciled so we good.